resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -80,40 +80,39 @@ protected void initialize(Class type, Type genericType, String paramName, Class
       baseType = type;
       baseGenericType = genericType;
 
-      if (type.isArray()) baseType = type.getComponentType();
-      if (List.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = ArrayList.class;
-      }
-      else if (SortedSet.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = TreeSet.class;
-      }
-      else if (Set.class.isAssignableFrom(type))
+      if (type.isArray())
       {
-         isCollection = true;
-         collectionType = HashSet.class;
+    	  baseType = type.getComponentType();
       }
-      if (isCollection)
+     
+      if (!baseType.isPrimitive())
       {
-         if (genericType != null && genericType instanceof ParameterizedType)
+         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
+         if (paramConverter != null) 
          {
-            ParameterizedType zType = (ParameterizedType) genericType;
-            baseType = Types.getRawType(zType.getActualTypeArguments()[0]);
-            baseGenericType = zType.getActualTypeArguments()[0];

```

Comment:
To my mind baseGenericType is modified from it's original value in case of we are dealing with a collection because in this case the `factory.getParamConverter(...)` must be in charge of finding a converter for the data type held in the collection and not for the collection type itself any longer since the call to this method on line 90 return null.
Whatever the reason as you said if prefer to add call on line 90 rather than changing everything. However I've just noticed that I did it wrong cause I've changed the original behavior...
 The call from line 88 to 94 should  have been
`     if (!baseType.isPrimitive())
      {
         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
         if (paramConverter != null) 
         {
        	 return;
}
}`
Then remove code from line 110 to 114 and put it in a `if ` block like following at line 116:
`         if (!baseType.isPrimitive())
         {
	         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
	         if (paramConverter != null) 
	         {
	        	 return;
	         }
     unmarshaller = factory.createStringParameterUnmarshaller(baseType);`